sdk/storage/azure-storage-blob-batch/src/test/java/com/azure/storage/blob/batch/BatchAPITest.groovy

```diff
@@ -128,14 +152,19 @@ class BatchAPITest extends APISpec {
         batchClient.submitBatchWithResponse(batch, false, null, Context.NONE)
 
         then:
-        notThrown(BlobStorageException)
-        response1.getStatusCode() == 200
+        notThrown(BlobBatchStorageException)
 
-        when:
-        response2.getStatusCode()
-
-        then:
-        thrown(BlobStorageException)
+        // In PLAYBACK check responses in an order invariant fashion.
+        if (testMode == TestMode.PLAYBACK) {
+            assertExpectedOrException(response1, 200) || assertExpectedOrException(response2, 200)
```

Comment:
If there is definitely one will fail, it is better to check fail at &.
```
!assertExpectedOrException(response1, 200) && assertExpectedOrException(response2, 200)
```

Reply:
The code would need to be the following, which is equivalent to what is already being used.

```java
!(assert... && assert...)
```

To be more pedantic on validation I could explicitly check each of these for being `true` and asserting the count of trues to be one. Another option is having these return `0` for failure and `1` for success and adding and checking it equals `1`.

Reply:
Updated to doing a count validation on successful responses.